### PR TITLE
Updates for Spec v0.10.0

### PR DESCRIPTION
**Pull request by wilwade**
Problem
=======
SDK needed to be updated to the new version of the spec
[#179299786](https://www.pivotaltracker.com/story/show/179299786)

Solution
========
Updated serialization for decimal and hexadecimal values and Parquet Schema changes

Double Checks:
---------------
- [x] Did you update the changelog?
- [x] Any new modules need to be exported?
- [x] Are new methods in the right module?
- [x] Are new methods/modules in core or not (i.e. porcelain or plumbing)?
- [x] Do you have good documentation on exported methods?

Change summary:
---------------
- Use BigInt for DSNPUserId
- Store byte data in parquet as bytes
- Parquet Schema changes
- Use Spec v0.10.0 serialization for signatures

### CODE REVIEW

File: src/core/announcements/serialization.ts [hunk 1]
```diff
@@ -17,50 +14,9 @@ export const serialize = (announcement: Announcement): string => {
...
 
-/**
- * serializeToHex() takes in just about anything and attempts to coerce the value to DSNP hexadecimal standard
- *
- * @param value - DSNP User Id/URI BigNumber BigInt number and even unknown
- * @throws {@link InvalidHexadecimalSerialization}
- * Thrown if the coercion failed
- * @returns A hexadecimal serialized value
- */
-export const serializeToHex = (
```
Comment (wilwade) [inline src/core/announcements/serialization.ts:28]: This is one of the big changes. The hex serialization is no longer needed as we are not trying to serialize a wide range of values and strip leading zeros.

This is mostly replaced not by `serializeValue`, but `convertToDSNPUserId` and `convertToDSNPUserURI` from #158

File: src/core/batch/batch.ts [hunk 2]
```diff
@@ -51,6 +51,21 @@ const parseAnnouncement = <T extends SignedAnnouncement>(record: ParquetRecord):
   return announcement as unknown as T;
 };
 
+const encodeParquetRecord = <T extends AnnouncementType>(announcement: AnnouncementWithSignature<T>): ParquetRecord => {
```
Comment (wilwade) [inline src/core/batch/batch.ts:54]: This makes sure the values going into Parquet are correct for parquetjs.

Aka most of it is supported out of the box, but hex as byte arrays is not. So this translates those hex values into Uint8Arrays which are supported. `parseAnnouncement` goes the other way back out.

File: src/core/batch/buffers.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
+import { HexString } from "../../types/Strings";
+
```
Comment (wilwade) [inline src/core/batch/buffers.ts:2]: The `uint8ArrayToHex` is from https://blog.xaymar.com/2020/12/08/fastest-uint8array-to-hex-string-conversion-in-javascript/ and the `hexToUint8Array` is just reversing it.

File: src/core/identifiers/identifiers.ts [hunk 4]
```diff
@@ -17,15 +19,15 @@ export type DSNPAnnouncementURI = string;
...
+  return id.match(/^dsnp:\/\/[0-9]{1,20}\/0x[0-9a-f]{64}$/) !== null;
 };
 
 /**
  * DSNPUserId represents a DSNP user id following the DSNP
  * [Identifiers](https://github.com/LibertyDSNP/spec/blob/main/pages/Identifiers.md)
  * specification.
  */
-export type DSNPUserId = string;
+export type DSNPUserId = bigint;
```
Comment (wilwade) [inline src/core/identifiers/identifiers.ts:30]: Important change!

**Review by enddynayn — APPROVED** [review]

File: src/core/batch/buffers.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+ * @param arr - The array or buffer to translate to hexadecimal
+ * @returns The hexadecimal string with a 0x prefix
+ */
+export const uint8ArrayToHex = (arr: Uint8Array | Buffer): HexString => {
+  let out = "";
+  for (let idx = 0, edx = arr.length; idx < edx; idx++) {
+    out += LUT_HEX_8B[arr[idx]];
+  }
+  return `0x${out}`;
+};
```
Comment (rosalinekarr) [inline src/core/batch/buffers.ts:38]: I wonder if these functions will ever be used elsewhere, and if so, should we move them to the utilities module instead?

**Review by rosalinekarr — APPROVED** [review]
One non-blocking thought, but otherwise LGTM

File: src/core/batch/buffers.ts (hunk 5 above)
Comment (wilwade) [reply src/core/batch/buffers.ts:38]: I would be surprised if they were?

I intentionally didn't export them from the batch module so they couldn't be used by normal sdk users.

**Pull request merged**